tests/unittests/test_timeout.py

```diff
@@ -0,0 +1,747 @@
+        # Set asana client in Context before test
+        Context.asana = Asana('test', 'test', 'test', 'test', 'test')
+        # initialize config file
+        Context.config = {'request_timeout': ''}
+        Context.asana.client.sections = mock.MagicMock()
+        Context.asana.client.sections.get_sections_for_project.side_effect = valid_data
+
+        sections_for_projects = sections.get_sections_for_projects('test', 'test', ['test'])
+
+        # verify if the timeout was passed as expected from the config file
+        Context.asana.client.sections.get_sections_for_project.assert_called_with(opt_fields=['test'], owner='test', project_gid='test', timeout=300.0)
+
+    def test_timeout_0_value_in_config__get_sections_for_projects(self, mocked_sleep, mocked_refresh_access_token):
+        # Set asana client in Context before test
+        Context.asana = Asana('test', 'test', 'test', 'test', 'test')
+        # initialize config file
+        Context.config = {'request_timeout': 0.0}
+        Context.asana.client.sections = mock.MagicMock()
+        Context.asana.client.sections.get_sections_for_project.side_effect = valid_data
+
+        sections_for_projects = sections.get_sections_for_projects('test', 'test', ['test'])
+
+        # verify if the timeout was passed as expected from the config file
+        Context.asana.client.sections.get_sections_for_project.assert_called_with(opt_fields=['test'], owner='test', project_gid='test', timeout=300.0)
+
+    def test_timeout_string_0_value_in_config__get_sections_for_projects(self, mocked_sleep, mocked_refresh_access_token):
+        # Set asana client in Context before test
+        Context.asana = Asana('test', 'test', 'test', 'test', 'test')
+        # initialize config file
+        Context.config = {'request_timeout': '0.0'}
+        Context.asana.client.sections = mock.MagicMock()
+        Context.asana.client.sections.get_sections_for_project.side_effect = valid_data
+
+        sections_for_projects = sections.get_sections_for_projects('test', 'test', ['test'])
+
+        # verify if the timeout was passed as expected from the config file
+        Context.asana.client.sections.get_sections_for_project.assert_called_with(opt_fields=['test'], owner='test', project_gid='test', timeout=300.0)
+
+@mock.patch("tap_asana.asana.Asana.refresh_access_token")
+@mock.patch("time.sleep")
+class TestTimeoutErrorStories(unittest.TestCase):
+
+    def test_timeout_error__get_stories_for_tasks(self, mocked_sleep, mocked_refresh_access_token):
+        # Set asana client in Context before test
+        Context.asana = Asana('test', 'test', 'test', 'test', 'test')
+        Context.asana.client.stories = mock.MagicMock()
+        Context.asana.client.stories.get_stories_for_task.side_effect = raise_Timeout_error
+
+        try:
+            stories_for_tasks = stories.get_stories_for_tasks('test', ['test'])
+        except requests.Timeout:
```

Comment:
Please explain the reason for the exception here. Mention it is intentional.

Reply:
Added comment